Horizons-Games/Axolotl-Engine
# feat: First version Hacking

Here we go with hacking ability! Main features:

### Hacking Zone
When the player reaches a certain influence area (GO with Rb + HackZone script), there is the possibility to start the hacking by pressing 'H'. 
- Auto-generated combination of buttons each time you start the hacking.
- Time limit to complete.
- Config options: influence radius, max time to complete and hacking sequence size.
  - Due to UI limitations, now there is a max sequence of 3 (the number of placeholders in the Hud, but can be easily extendable).

### Hacking Combination
- The list of hacking inputs is in HackingCommand script along with the proper mappings to user input type. 
- The UIHackingManager is inspired by the combat one, having the GOs of the commands there, linked to the hacking command mentioned. 
  - Now only two options can be linked (X,Y). UI should incorporate the rest of the controller button images.
- So, if any key/button needs to be readjusted the only thing to do is add the mapping and the GOs to the UI script.

### Steps to test the hacking

1. Copy/Paste the CombatSystem UI element in the hierarchy, delete the Bar object and leave the same enabled/disabled states of the children as the original CombatSystem. Replace the script of UICombo for the UIHackingManager and drag the placeholders (X, Y).
2. Add PlayerHackingUseScript to the character and attach the duplicated UI element above.
3. Create a GameObject in the scene with a RigidBody and the HackZone script.
4. Play with Keyboard or Controller, but for enabling the hacking use the "H" key. Then the combination will appear as buttons, but "translated" in keys in the engine console (if you want to test it with keyboard).
 
### Missing functionalities
- UI should be improved, with more images of the buttons and a proper canvas to display a sequence as long as we want.
- Enabling/Disabling the environment. ComponentAgents should be all disabled as well as Player's RigidBody.
